home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/cooperations/CooperationApiIT.java

```diff
@@ -0,0 +1,186 @@
+import java.util.List;
+import com.softserveinc.ita.homeproject.ApiException;
+import com.softserveinc.ita.homeproject.ApiResponse;
+import com.softserveinc.ita.homeproject.api.CooperationApi;
+import com.softserveinc.ita.homeproject.api.tests.utils.ApiClientUtil;
+import com.softserveinc.ita.homeproject.model.Address;
+import com.softserveinc.ita.homeproject.model.CreateCooperation;
+import com.softserveinc.ita.homeproject.model.CreateHouse;
+import com.softserveinc.ita.homeproject.model.ReadCooperation;
+import com.softserveinc.ita.homeproject.model.UpdateCooperation;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.Test;
+
+class CooperationApiIT {
+
+    private final CooperationApi cooperationApi = new CooperationApi(ApiClientUtil.getClient());
+
+    @Test
+    void createCooperationTest() throws ApiException {
+        CreateCooperation createCoop = createCooperation();
+        ApiResponse<ReadCooperation> response = cooperationApi.createCooperationWithHttpInfo(createCoop);
+        assertEquals(Response.Status.CREATED.getStatusCode(), response.getStatusCode());
+        assertCooperation(createCoop, response.getData());
+    }
+
+    @Test
+    void getCooperationTest() throws ApiException {
+        CreateCooperation createCoop = createCooperation();
+        ReadCooperation savedCooperation = cooperationApi.createCooperation(createCoop);
+        ApiResponse<ReadCooperation> response = cooperationApi.getCooperationWithHttpInfo(savedCooperation.getId());
+        assertEquals(Response.Status.OK.getStatusCode(), response.getStatusCode());
+        assertCooperation(createCoop, response.getData());
+    }
+
+    @Test
+    void updateCooperationTest() throws ApiException {
+        CreateCooperation createCoop = createCooperation();
+        ReadCooperation savedCooperation = cooperationApi.createCooperation(createCoop);
+
+        UpdateCooperation updateCoop = new UpdateCooperation()
+            .name("upd")
+            .usreo(RandomStringUtils.randomAlphabetic(10))
+            .iban(RandomStringUtils.randomAlphabetic(20))
+            .address(new Address()
+                .city("upd")
+                .district("upd")
+                .houseBlock("upd")
+                .houseNumber("upd")
+                .region("upd")
+                .street("upd")
+                .zipCode("upd"));
+
+        ApiResponse<ReadCooperation> response = cooperationApi
+            .updateCooperationWithHttpInfo(savedCooperation.getId(), updateCoop);
+
+        assertEquals(Response.Status.OK.getStatusCode(), response.getStatusCode());
+        assertCooperation(savedCooperation, updateCoop, response.getData());
+    }
+
+    @Test
+    void deleteCooperationTest() throws ApiException {
+        ReadCooperation readCoop = cooperationApi.createCooperation(createCooperation());
+        ApiException exception = new ApiException(404, "Can't find cooperation with given ID: " + readCoop.getId());
```

Comment:
Why do we need to build this object? Can we just assert the message?

Reply:
Done.